controller/aks-cluster-config-handler.go

```diff
@@ -668,17 +668,31 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, secretsCache w
 
 		// check for updated NodePools
 		upstreamNodePools, _ := utils.BuildNodePoolMap(upstreamSpec.NodePools, config.Spec.ClusterName)
-		updateNodePool := false
 		for npName, np := range downstreamNodePools {
+			updateNodePool := false
 			upstreamNodePool, ok := upstreamNodePools[npName]
 			if ok {
-				// There is a matching node pool in the cluster already, so update it if needed
-				if to.Int32(np.Count) != to.Int32(upstreamNodePool.Count) {
+				if to.Bool(np.EnableAutoScaling) && to.Int32(np.Count) != to.Int32(upstreamNodePool.Count) {
+					return config, fmt.Errorf("can't update node count when autoscaling is enabled in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
+				}
+				if !to.Bool(np.EnableAutoScaling) && to.Int32(np.Count) != to.Int32(upstreamNodePool.Count) {
 					logrus.Infof("Updating node count in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
 					updateNodePool = true
 				}
-				if np.EnableAutoScaling != nil && to.Bool(np.EnableAutoScaling) != to.Bool(upstreamNodePool.EnableAutoScaling) {
-					logrus.Infof("Updating autoscaling in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
+				if to.Bool(np.EnableAutoScaling) && to.Bool(np.EnableAutoScaling) != to.Bool(upstreamNodePool.EnableAutoScaling) {
+					logrus.Infof("Enable autoscaling in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
+					updateNodePool = true
+				}
+				if to.Bool(np.EnableAutoScaling) && to.Int32(np.MinCount) != to.Int32(upstreamNodePool.MinCount) {
+					logrus.Infof("Updating minimum count in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
+					updateNodePool = true
+				}
+				if to.Bool(np.EnableAutoScaling) && to.Int32(np.MaxCount) != to.Int32(upstreamNodePool.MaxCount) {
+					logrus.Infof("Updating maximum count in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
+					updateNodePool = true
+				}
```

Comment:
Should there be a validation error if `np.EnableAutoScaling` is false but the user tries to update MinCount or MaxCount? It look like it's just ignored.

Reply:
added check for this case